### PR TITLE
Insert-Delete orderedArray.py

### PR DESCRIPTION
**Pull request by ShiraSchneid**
Issue #35 
Shani and I added  insert and remove functions to Ordered Array. We took most of the remove function from Array.py. In the find function, we also changed the return value to be the index of mid as opposed to the value. And, we changed clickFind to use errors based off the change in find. We also changed two lines in assignElements to remove "fromIndex" so the animation would remove duplicated cells. This is the most current version of OrderedArray.py. 
You can find it here: https://github.com/JMCanning78/datastructures-visualization/pull/48/commits/aefee6ea8a5ca82572ea6209af019861a3f4eef4

### CODE REVIEW

**Comment by ShiraSchneid** [comment]
We found an issue with inserting two things in a row. We're working on it.

**Comment by JMCanning78** [comment]
There's a link near the top right of the PR detail page that says "Convert to draft".  You can use that if the PR is not ready for review.
When you fix the issues and want reviews, select "Ready for review"

File: PythonVisualizations/orderedArray.py [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+        j = self.find(val)  # Find where item should go
+
+        self.list.append(drawable(None))
+
+        for k in range(len(self.list) - 1, j, -1):  # Move bigger items right
+            self.wait(1)
+            self.list[k].val = self.list[k - 1].val
+            self.assignElement(k - 1, k)
+
+>>>>>>> Stashed changes
```
Comment (JMCanning78) [inline PythonVisualizations/orderedArray.py:110]: You must reconcile any merge conflicts, like the one here, before committing your code.  You can find them by searching for `====`.

File: PythonVisualizations/orderedArray.py [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+=======
+        self.window.update()       
+        
+    def remove(self,val):
+        self.startAnimations()
+        index = self.find(val)
+        if index != None:
+            self.wait(1)
+            self.cleanUp()
+>>>>>>> Stashed changes
```
Comment (JMCanning78) [inline PythonVisualizations/orderedArray.py:144]: Another merge conflict that must be resolved.

File: PythonVisualizations/orderedArray.py [hunk 3]
```diff
@@ -0,0 +1,491 @@
...
+<<<<<<< Updated upstream
+        
+        self.cleanUp(callEnviron)
+        return found
+=======
+            self.stopAnimations()
+            return True
+        self.stopAnimations()
+        return False 
+>>>>>>> Stashed changes
```
Comment (JMCanning78) [inline PythonVisualizations/orderedArray.py:171]: Another merge conflict that must be resolved.

File: PythonVisualizations/orderedArray.py [hunk 4]
```diff
@@ -0,0 +1,491 @@
...
+        n = self.list.pop()
+
+        # delete the associated display objects
+        self.canvas.delete(n.display_shape)
+        self.canvas.delete(n.display_val)
+        
+<<<<<<< Updated upstream
+=======
+        print(len(self.list))
+>>>>>>> Stashed changes
```
Comment (JMCanning78) [inline PythonVisualizations/orderedArray.py:215]: Another merge conflict.  The print statement is probably for debugging purposes and should be removed in the version that is committed.

File: PythonVisualizations/orderedArray.py [hunk 5]
```diff
@@ -0,0 +1,491 @@
...
+        self.canvas.delete(self.list[fromIndex].display_val)
+        self.canvas.delete(self.list[fromIndex].display_shape)           
+        
+=======
+        
+        # delete the original "from" display value and the new display shape
+        self.canvas.delete(self.list[fromIndex].display_val)
+        self.canvas.delete(self.list[fromIndex].display_shape)        
+
+>>>>>>> Stashed changes
```
Comment (JMCanning78) [inline PythonVisualizations/orderedArray.py:260]: Another merge conflict

File: PythonVisualizations/orderedArray.py [hunk 6]
```diff
@@ -0,0 +1,491 @@
...
+        lo = 0                             #Point to lo
+        indexLo = self.createIndex(lo, 'lo',level= 1)
+        self.cleanup |= set(indexLo)
+        hi = len(self.list)-1              # Point to hi
+        indexHi = self.createIndex(hi, 'hi', level = 3)
+        self.cleanup |= set(indexHi)
+        mid = (lo + hi) // 2               # Point to the midpoint
+        indexMid = self.createIndex(mid, 'mid', level = 2)
+        self.cleanup |= set(indexMid)            
+>>>>>>> Stashed changes
```
Comment (JMCanning78) [inline PythonVisualizations/orderedArray.py:356]: Another merge conflict

File: PythonVisualizations/orderedArray.py [hunk 7]
```diff
@@ -0,0 +1,491 @@
...
+            "Random Fill", lambda: self.randomFill())
+        deleteRightmostButton = self.addOperation(
+            "Delete Rightmost", lambda: self.removeFromEnd())
+        #this makes the pause, play and stop buttons 
+<<<<<<< Updated upstream
+        self.addAnimationButtons()
+        
+=======
+        self.addAnimationButtons()        
+>>>>>>> Stashed changes
```
Comment (JMCanning78) [inline PythonVisualizations/orderedArray.py:408]: Another merge conflict

File: PythonVisualizations/orderedArray.py [hunk 8]
```diff
@@ -0,0 +1,491 @@
...
+            
+    def stopAnimations(self):
+        super().stopAnimations()
+        self.enableButtons(enable=True)
+<<<<<<< Updated upstream
+        self.argumentChanged()    
+    
+=======
+        self.argumentChanged()        
+>>>>>>> Stashed changes
```
Comment (JMCanning78) [inline PythonVisualizations/orderedArray.py:485]: Another merge conflict

**Comment by ShiraSchneid** [comment]
June 29 Update
Shani and I made sure that OrderedArray was up to date with all GitHub merges and updates. This code has all the basic function (new array size, fill the array, deleteRightMost, find) and insert and delete. The animation buttons are also included.

**Comment by shirapahmer** [comment]
We tried running the orderedArray.py and got an error that we think relates to the fact that the branch hasn't merged from the master. It says:
  ```
File "/Users/pahmer/Documents/GitHub/datastructures-visualization/PythonVisualizations/orderedArray.py", line 199, in createCellValue
    *valPos, text=str(key), font=self.VALUE_FONT, fill=self.VALUE_COLOR)
AttributeError: 'OrderedArray' object has no attribute 'VALUE_FONT'
```
Should we wait to test the code until it has been merged or should it really be running for us anyway and the problem is on our end and not from the orderedArray.py itself?

**Pull request merged**